workaround #378: generate a valid sarif file when no files are scanned

andyjones opened the pull request:
I use semgrep in github actions and upload the sarif file to Github. The job sometimes fails with a schema validation error when no files are scanned as described in #378 and #421 

This tweaks the output to generate a minimum sarif file that validates against the 2.1.0 json schema.

### Security

- [ ] Change has no security implications (otherwise, ping the security team)

andyjones committed:
workaround #378: generate a valid sarif file when no files are scanned

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/returntocorp/semgrep-action?pullRequest=457) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/returntocorp/semgrep-action?pullRequest=457) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/returntocorp/semgrep-action?pullRequest=457) it.</sub>

brendongo commented:
Sarif spec copied here matches https://github.com/oasis-tcs/sarif-spec/blob/master/Documents/CommitteeSpecifications/2.1.0/sarif-schema-2.1.0.json but looks like that spec doesn't match the spec published on https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317736 specifically the suppression object has `state` property in the former but is `status` in the later. Going to copy over the spec from https://docs.oasis-open.org/sarif/sarif/v2.1.0/cos02/schemas/sarif-schema-2.1.0.json

brendongo commented on tests/unit/test_sarif.py:
```diff
@@ -0,0 +1,28 @@
+    return schema
+
+
+def test_rewrite_empty_sarif_file(tmp_path: Path, sarif_schema):
```
Thanks for the test!

brendongo committed:
Update sarif-schema-2.1.0.json

Copy the sarif schema from https://docs.oasis-open.org/sarif/sarif/v2.1.0/cos02/schemas/sarif-schema-2.1.0.json which had minor differences than the one found in https://github.com/oasis-tcs/sarif-spec/blob/master/Documents/CommitteeSpecifications/2.1.0/sarif-schema-2.1.0.json but matches the published plain text spec more

brendongo reviewed (approved):
Thanks for the contribution @andyjones !

andyjones commented:
thanks @brendongo , I missed that! FYI in case it helps others, github use a slightly different/more lax schema when they validate the upload as well: https://github.com/github/codeql-action/commits/main/src/sarif_v2.1.0_schema.json
